Build fails in Android Studio (Gradle 8.1.1 and JDK 21)

### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/AntennaPod/AntennaPod/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

NA

### Where did you get the app from

Other

### Android version

NA

### Device model

_No response_

### First occurred

_No response_

### Steps to reproduce

The debug build (both in the develop and master branch) fails with the current Android Studio:

    Android Studio Ladybug | 2024.2.1 Patch 2
    Build #AI-242.23339.11.2421.12550806, built on October 24, 2024
    Runtime version: 21.0.3+-12282718-b509.11 amd64

See build errors below.

The reason could be some JDK 17 / JDK 21 incompatibility.

Android Studio suggests running the "AGP Upgrade Assistant" to migrate from Gradle 8.1.1 to 8.7.2.
This solves the build problem with JDK 21. I have not tested it with JDK 17.



Build fails with errors:

    FAILURE: Build completed with 4 failures.

    1: Task failed with an exception.
    -----------
    * What went wrong:
    Execution failed for task ':model:compileDebugJavaWithJavac'.
    > Could not resolve all files for configuration ':model:androidJdkImage'.
       > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
          > Execution failed for JdkImageTransform: /home/willem/opt/android-sdk-linux/platforms/android-34/core-for-system-modules.jar.
             > Error while executing process /home/willem/opt/android-studio/jbr/bin/jlink with arguments {--module-path /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/temp/jmod --add-modules java.base --output /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/jdkImage --disable-plugin system-modules}

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.
    > Get more help at https://help.gradle.org.
    ==============================================================================

    2: Task failed with an exception.
    -----------
    * What went wrong:
    Execution failed for task ':net:ssl:compileFreeDebugJavaWithJavac'.
    > Could not resolve all files for configuration ':net:ssl:androidJdkImage'.
       > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
          > Execution failed for JdkImageTransform: /home/willem/opt/android-sdk-linux/platforms/android-34/core-for-system-modules.jar.
             > Error while executing process /home/willem/opt/android-studio/jbr/bin/jlink with arguments {--module-path /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/temp/jmod --add-modules java.base --output /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/jdkImage --disable-plugin system-modules}

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.
    > Get more help at https://help.gradle.org.
    ==============================================================================

    3: Task failed with an exception.
    -----------
    * What went wrong:
    Execution failed for task ':ui:i18n:compileDebugJavaWithJavac'.
    > Could not resolve all files for configuration ':ui:i18n:androidJdkImage'.
       > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
          > Execution failed for JdkImageTransform: /home/willem/opt/android-sdk-linux/platforms/android-34/core-for-system-modules.jar.
             > Error while executing process /home/willem/opt/android-studio/jbr/bin/jlink with arguments {--module-path /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/temp/jmod --add-modules java.base --output /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/jdkImage --disable-plugin system-modules}

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.
    > Get more help at https://help.gradle.org.
    ==============================================================================

    4: Task failed with an exception.
    -----------
    * What went wrong:
    Execution failed for task ':ui:app-start-intent:compileDebugJavaWithJavac'.
    > Could not resolve all files for configuration ':ui:app-start-intent:androidJdkImage'.
       > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
          > Execution failed for JdkImageTransform: /home/willem/opt/android-sdk-linux/platforms/android-34/core-for-system-modules.jar.
             > Error while executing process /home/willem/opt/android-studio/jbr/bin/jlink with arguments {--module-path /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/temp/jmod --add-modules java.base --output /home/willem/.gradle/caches/transforms-4/9a7ee21472c198d770ae60685527dcce-324af383-23dc-4a33-be1b-155c2ccba6b9/transformed/output/jdkImage --disable-plugin system-modules}

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.
    > Get more help at https://help.gradle.org.
    ==============================================================================

    Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

    You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

    For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

    BUILD FAILED in 41s


### Expected behaviour

_No response_

### Current behaviour

_No response_

### Logs

_No response_

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -178,7 +178,8 @@
             android:configChanges="keyboardHidden|orientation|screenSize|screenLayout|smallestScreenSize"
             android:supportsPictureInPicture="true"
             android:screenOrientation="sensorLandscape"
-            android:exported="false">
+            android:exported="false"
+            tools:ignore="DiscouragedApi">
             <meta-data
                 android:name="android.support.PARENT_ACTIVITY"
                 android:value="de.danoeh.antennapod.activity.MainActivity"/>

--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.agpVersion = "8.1.1"
+    ext.agpVersion = "8.5.2"
 }
 plugins {
     id 'com.android.application' version "$agpVersion" apply false

--- a/common.gradle
+++ b/common.gradle
@@ -54,6 +54,7 @@ android {
 
     buildFeatures {
         viewBinding true
+        buildConfig true
     }
 }
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,3 @@
 android.useAndroidX=true
 org.gradle.jvmargs=-Xmx4096m
-android.defaults.buildfeatures.buildconfig=true
 android.nonTransitiveRClass=false

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.10-bin.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

--- a/ui/i18n/src/main/res/values-ja/strings.xml
+++ b/ui/i18n/src/main/res/values-ja/strings.xml
@@ -217,8 +217,6 @@
     <string name="download_running">ダウンロード実行中</string>
     <string name="download_error_details">詳細</string>
     <string name="download_log_details_message">%1$s \n\n技術的な理由: \n%2$s \n\nファイル URL:\n%3$s</string>
-    <string name="download_error_retrying">\\%1$s\\ のダウンロードに失敗しました。後ほど再度ダウンロードを試みます。</string>
-    <string name="download_error_not_retrying">\\%1$s\\ のダウンロードに失敗しました。</string>
     <string name="download_error_tap_for_details">タップして詳細を表示します。</string>
     <string name="download_error_device_not_found">ストレージ デバイスが見つかりません</string>
     <string name="download_error_insufficient_space">デバイスに十分な空きスペースがありません。</string>
@@ -465,7 +463,6 @@
   <string name="search_status_no_results">見つかりませんでした</string>
     <string name="type_to_search">検索ワードを入力してください</string>
     <string name="search_label">検索</string>
-    <string name="no_results_for_query">\\%1$s\\ の結果は見つかりませんでした</string>
     <string name="search_online">オンラインで検索</string>
     <!--Synchronization-->
   <string name="sync_status_started">同期開始</string>

